exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/StringFunctions.java

```diff
@@ -1425,7 +1419,7 @@ public void eval() {
       out.buffer = buffer = buffer.reallocIfNeeded((left.end - left.start) + (right.end - right.start));
       out.start = out.end = 0;
 
-      int id = 0;
+      int id;
```

Comment:
Nit: thanks for removing the unneeded initializer. Might as well move the type into the `for` loop:

```
for (int id =...
```

Here and below.

Reply:
In this case, the id variable is also used in the next loop, as a pointer to where to continue writing

Reply:
Fixed